starlite/dto/interface.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from abc import ABCMeta, abstractmethod
+from typing import TYPE_CHECKING, Any, Generic, TypeVar
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from starlite.connection import Request
+    from starlite.handlers import BaseRouteHandler
+    from starlite.types import StarliteEncodableType
+    from starlite.types.parsed_signature import ParsedType
+
+__all__ = ("AbstractDTOInterface", "DataT")
+
+DataT = TypeVar("DataT")
+"""Type var representing data held by a DTO instance."""
+
+
+class AbstractDTOInterface(Generic[DataT], metaclass=ABCMeta):
```

Comment:
this abstract class has no logic. Why not have it as a Protocol instead? You call it `Intefarce`, but the semantic name for this in python is `Protocol`

If not, please remove the work `Interface` and lets stick to the standard semantics of `AbstractDTO`.

Also, the rest of the codebase has a `base.py` file, not `interface.py`